app/controllers/sensor_readings_controller.rb

```diff
@@ -48,8 +48,51 @@ def fake
     end
   end
 
+  def add
+    if manual_params[:sensor_id].nil? || manual_params[:report_id].nil?
```

Comment:
Anything with `:sensor_id` in the `bin/rails routes` will also never be nil. You can require certein params with strong params: https://github.com/rails/strong_parameters